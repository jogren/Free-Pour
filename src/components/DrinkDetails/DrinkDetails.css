.Details_section {
  position: fixed;
  left: 17%;
  background-color: #FFF;
  opacity: 0.95;
  width: 70%;
  height: 500px;
  overflow: scroll;
  margin-top: 20px;
}

.Details_button-container {
  width: 100%;
  display: flex;
  justify-content: flex-end;
}

.Details_section img {
  height: 200px;
}

.Details_img-ingredients {
  display: flex;
  justify-content: space-around;
  margin-bottom: 15px;
}

.Details_section p {
  text-align: center;
}

.Details-name {
  font-size: 42px;
}

.Details-glass {
  font-size: 18px;
  margin-bottom: 10px;
}

.Details-ingredients {
  padding: 10px;
}

.Details-ingredients p {
  font-size: 18px;
  margin-bottom: 3px;
}

.Details-instructions {
  margin: 0px 15px;
}

h6 {
  font-size: 24px;
  padding-bottom: 3px;
  border-bottom: 3px solid black;
  margin-bottom: 3px;
  text-align: center;
}
