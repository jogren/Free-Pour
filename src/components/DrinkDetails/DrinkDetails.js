import React from 'react';
import './DrinkDetails.css';

const DrinkDetails = ({ selectedDrink, hideSelectedDrink}) => {
  let ingredientsList = selectedDrink.ingredients.map((ingredient, index) => {
    if (ingredient.measure !== '' && ingredient.ingredient !== '') {
      return <p key={index}>{ingredient.measure} {ingredient.ingredient}</p>
    }
  });

  return (
    <section className="Details_section">
      <div className="Details_button-container">
        <button onClick={hideSelectedDrink}>x</button>
      </div>
      <p className="Details-name">{selectedDrink.name}</p>
      <p className="Details-glass">{selectedDrink.glass}</p>
      <div className="Details_img-ingredients">
        <img src={selectedDrink.image} alt={selectedDrink.name}/>
        <div className="Details-ingredients">
          <h6>Ingredients!</h6>
          {ingredientsList}
        </div>
      </div>
      <p className="Details-instructions">{selectedDrink.instructions}</p>
    </section>
  );
}

export default DrinkDetails;