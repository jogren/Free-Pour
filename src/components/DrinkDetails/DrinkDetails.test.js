import React from 'react';
import { shallow } from 'enzyme';
import DrinkDetails from './DrinkDetails';

describe('DrinkDetails', () => {
  let wrapper;
  const mockHideSelectedDrink = jest.fn();
  const mockSelectedDrink = {
    name: 'Mojito',
    glass: 'Highball glass',
    instructions: 'Muddle mint leaves...',
    image: 'image.png',
    ingredients: [
      {
        guessed: false,
        ingredient: 'Light rum',
        measure: '2-3 oz'
      },
      {
        guessed: false,
        ingredient: 'mint',
        measure: 'leaves'
      },
      {
        guessed: false,
        ingredient: 'sugar',
        measure: '2tbsp'
      }
    ]
  }

  beforeEach(() => {
    wrapper = shallow(<DrinkDetails 
      selectedDrink={mockSelectedDrink}
      hideSelectedDrink={mockHideSelectedDrink}
    />)
  })

  it('should take a snapshot when passed the correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });
})