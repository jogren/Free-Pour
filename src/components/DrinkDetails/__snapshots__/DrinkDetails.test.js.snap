// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrinkDetails should take a snapshot when passed the correct data 1`] = `
<section
  className="Details_section"
>
  <div
    className="Details_button-container"
  >
    <button
      onClick={[MockFunction]}
    >
      x
    </button>
  </div>
  <p
    className="Details-name"
  >
    Mojito
  </p>
  <p
    className="Details-glass"
  >
    Highball glass
  </p>
  <div
    className="Details_img-ingredients"
  >
    <img
      alt="Mojito"
      src="image.png"
    />
    <div
      className="Details-ingredients"
    >
      <h6>
        Ingredients!
      </h6>
      <p
        key="0"
      >
        2-3 oz
         
        Light rum
      </p>
      <p
        key="1"
      >
        leaves
         
        mint
      </p>
      <p
        key="2"
      >
        2tbsp
         
        sugar
      </p>
    </div>
  </div>
  <p
    className="Details-instructions"
  >
    Muddle mint leaves...
  </p>
</section>
`;
